### PR TITLE
job analyser

### PR DESCRIPTION
**Pull request by FrostGod**
Added the basic structure of the job analyser 
{'input': 'fetch the job description and then using your intelligence list the top 3 key skills with some importance out of 10 required for this job?', 'output': '1. Python (9/10)\n2. C/C++ (8/10) \n3. Experience with Gitlab build, test and automation (7/10)'}

### CODE REVIEW

File: backend/job-analyse.py
Comment (Stiffjobs) [inline backend/job-analyse.py:1]: Is it necessary to split the job description content? If the content is divided into two parts, retrieving the entire job description might become difficult. When trying to rank the top 50 suitable job descriptions for the current user's resume (experience and projects), we should evaluate them using the complete job description, right?

**Comment by Stiffjobs** [comment]
btw, should we organize our backend code into the functions directory to achieve a better project structure? The functions directory is intended for running cloud functions, allowing us to consolidate all of our python code there.

**Comment by Stiffjobs** [comment]
> Added the basic structure of the job analyser {'input': 'fetch the job description and then using your intelligence list the top 3 key skills with some importance out of 10 required for this job?', 'output': '1. Python (9/10)\n2. C/C++ (8/10) \n3. Experience with Gitlab build, test and automation (7/10)'}

**Comment by Stiffjobs** [comment]
[manage cloud functions](https://firebase.google.com/docs/functions/manage-functions?gen=2nd#set_python_version)

**Comment by FrostGod** [comment]
> btw, should we organize our backend code into the functions directory to achieve a better project structure? The functions directory is intended for running cloud functions, allowing us to consolidate all of our python code there.

Yeah no worries, I will move it there, this was just a rough draft. which just works for one job description at a time, i would merege this once the data pipeline is fetched for job description

File: backend/job-analyse.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    output_parser = StrOutputParser()
+
+    setup_and_retrieval = RunnableParallel(
+        {"context": retriever, "question": RunnablePassthrough()}
+    )
+    model = self.google_genai
+    chain = setup_and_retrieval | prompt | model
+
+    key_skills = chain.invoke("Using your intelligence, what is the suitable job title for this role and what are the top 3 key skills aligned with importance out of 10 required for this job?")
+    key_skills.content = key_skills.content[7:-3]
```
Comment (knhn1004) [inline backend/job-analyse.py:97]: what is the purpose of `7:-3`

Comment (FrostGod) [reply backend/job-analyse.py]: ahh the prompt output was like
"""json < our required content > """
I am just extracting the content

**Pull request merged**